### PR TITLE
chore: bump stackable-operator version and cargo update

### PR DESCRIPTION
**Pull request by xeniape**
# Description

PR to fix unmaintained crate cargo deny errors

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Changed
+
+- Bump `stackable-operator` from `0.80.0` to `0.83.0` ([#316]).
```
Comment (sbernauer) [inline CHANGELOG.md:9]: We normally don't document op-rs bumps in here - we only do so when there are user-facing changes

**Review by sbernauer — CHANGES_REQUESTED** [review]
Thanks for fixing them!

File: CHANGELOG.md (hunk 1 above)
Comment (xeniape) [reply CHANGELOG.md]: https://github.com/stackabletech/commons-operator/pull/316/commits/b0c95fd664d11b7bc86349c57f4eabe745b113a4

**Review by sbernauer — APPROVED** [review]
Thanks!

**Pull request merged**